aderyn_core/src/report/markdown_printer.rs

```diff
@@ -0,0 +1,222 @@
+                    .title
+                    .to_lowercase()
+                    .replace(' ', "-")
+                    .replace(|c: char| !c.is_ascii_alphanumeric() && c != '-', "");
+                writeln!(
+                    w,
+                    "  - [C-{}: {}](#C-{}-{})",
+                    index + 1,
+                    issue.title,
+                    index + 1,
+                    issue_title_slug
+                )
+                .unwrap();
+            }
+        }
+
+        fn display<T: Write>(title: &str, issues: &[Issue], mut writer: T) {
+            if !issues.is_empty() {
+                writeln!(writer, "{}", title).unwrap();
+                print_table_of_content(issues, &mut writer);
+            }
+        }
+
+        writeln!(writer, "# Table of Contents\n")?;
+        writeln!(writer, "- [Summary](#summary)")?;
+        writeln!(writer, "  - [Files Summary](#files-summary)")?;
+        writeln!(writer, "  - [Files Details](#files-details)")?;
+        writeln!(writer, "  - [Issue Summary](#issue-summary)")?;
+
+        let issues = [
+            (&report.criticals, "- [Critical Issues](#critical-issues)"),
+            (&report.highs, "- [High Issues](#high-issues)"),
+            (&report.mediums, "- [Medium Issues](#medium-issues)"),
+            (&report.lows, "- [Low Issues](#low-issues)"),
+            (&report.ncs, "- [NC Issues](#nc-issues)"),
+        ];
+
+        issues.iter().for_each(|rec| {
+            display(rec.1, rec.0, &mut writer);
+        });
+
+        writeln!(writer, "\n")?; // Add an extra newline for spacing
+        Ok(())
+    }
+
+    fn print_issue<W: Write>(
+        &self,
+        mut writer: W,
+        issue_body: &IssueBody,
+        severity: &str,
+        number: i32,
+        root_path: &Path,
+    ) -> Result<()> {
+        let is_file = root_path.is_file();
+
+        writeln!(
+            writer,
+            "## {}-{}: {}\n\n{}\n", // <a name> is the anchor for the issue title
+            severity, number, issue_body.title, issue_body.description
+        )?;
+        for instance in &issue_body.instances {
+            let path = {
+                if is_file {
+                    String::from(root_path.to_str().unwrap())
+                } else {
+                    String::from(
+                        root_path
+                            .join(instance.contract_path.clone())
+                            .as_path()
+                            .to_str()
+                            .unwrap(),
+                    )
+                }
+            };
+
+            let line = std::fs::read_to_string(&path).unwrap();
+
+            let line_preview = line
+                .split('\n')
+                .skip(instance.line_no - 1)
+                .take(1)
+                .next()
+                .unwrap();
+
+            writeln!(
+                writer,
+                "- Found in {} Line: {}\n\n\t>{}\n\nfile://{}\n\n\n",
```

Comment:
I'm wondering if we make the line preview use a code snippet format instead of a quote format.

So instead of:

>            target.delegatecall(abi.encodeWithSignature("doSomething(uint256)", I));

We print:
```solidity
            target.delegatecall(abi.encodeWithSignature("doSomething(uint256)", I));
```

Reply:
Yes, that seems to bring out the correct semantic meaning. You are right!